packages/@sanity/mutator/src/jsonpath/tokenize.ts

```diff
@@ -113,6 +113,19 @@ class Tokenizer {
       return null
     }
     if (str === this.source.slice(this.i, this.i + str.length)) {
+      // When checking symbols that consist of valid attribute characters, we
+      // need to make sure we don't inadvertently treat an attribute as a
+      // symbol. For example, an attribute 'trueCustomerField' should not be
+      // scanned as the boolean symbol "true".
+      if (str[0].match(attributeCharMatcher)) {
```

Comment:
do we know what the perf hit is on looking ahead on each match?

Reply:
It would only to this extra check when it encounters a symbol match which is all attribute-valid chars, meaning `"true"` and `"false"`